platforms/jvm/code-quality/src/main/java/org/gradle/api/plugins/quality/CodeQualityExtension.java

```diff
@@ -85,13 +89,53 @@ public void setIgnoreFailures(boolean ignoreFailures) {
      */
     @ToBeReplacedByLazyProperty
     public File getReportsDir() {
-        return reportsDir;
+        return maybeInitReportsDir().get();
     }
 
     /**
      * The directory where reports will be generated.
      */
     public void setReportsDir(File reportsDir) {
-        this.reportsDir = reportsDir;
+        maybeInitReportsDir().set(reportsDir);
+    }
+
+    /**
+     * Visible for internal use only!
+     *
+     * @return the reports directory property
+     * @since 9.0
+     */
+    @Incubating
+    public Property<File> getReportsDirProperty() {
+        return maybeInitReportsDir();
+    }
+
+    /**
+     * Visible for internal use only!
+     * @return the source sets property
+     * @since 9.0
+     */
+    @Incubating
+    public ListProperty<SourceSet> getSourceSetsProperty() {
+        return maybeInitSourceSets();
+    }
+
+    private Property<File> maybeInitReportsDir() {
```

Comment:
This "maybe" in the method names suggest some kind of conditional code, but it's not the case. The property will always be initialized, if it's not already. It's not like it will only be initialized if not already AND some other condition. It's confusing to me. I would just drop the "maybe" prefix.